README.md

```diff
@@ -1,87 +1,102 @@
+* `-f` the file we want the backend to store its database on. Default is "org.db" for an organizer and "wit.db" for a witness
 
-## Packages
+You can generate the code documentation using the godoc command: `godoc -http:=6060`.
 
+## Packages
 ### actors
-This package implements the actors defined by the PoP project, and their reactions to received messages.
+This package implements the actors defined by the PoP project (organizer and witness), and their reactions to received messages. 
+The "actor" interface defines what functions are mandatory for an actor to implement.
 
 ### db 
-This package's purpose is to handle interaction with databases. The databases structure is described in the section "Back-end" components.
+This package's purpose is to handle interaction with the local database. The database's structure is described in the section [Back-end components](#Database-structure).
 
 ### network 
-This package is to handle interactions with the websocket. It has a hub, which handles the publish-subscribe paradigm, and
-the hub has an actor (either Organizer or Witness), that implements the PoP concept.
+This package's purpose is to handle the webserver listening for websocket connections and to handle them.
+It has a type `Hub`, that decides what message to send to whom according to the publish-subscribe paradigm.
+The hub has an actor (either Organizer or Witness), that implements the PoP concept, and generate response messages 
+depending on the received message.
 
 ### event
-This package provides structure for "real world events" and corresponding jsonRPC messages,  methods to convert byte received arrays into theses structures
-and function to check validity of the messages and signatures we received.
+This package provides the structures for real world events. We use it when storing the different components internally.
+
+### message
+This package provides structures corresponding to the messages that can be sent or received as defined in the jsonRPC package.
 
 ### jsonRPC
-This package is not used at all, but is defines the protocol we use to format the messages we send and receive.
+This package is not directly used by this backend version, but defines the messaging protocol used to on the websockets.
 
 ### lib
-This package defines gloabal useful functions, like find an element in an array, etc.
-
-### message
-This package implements the messages that can be sent or received as defined in the jsonRPC package.
+This package defines global useful functions, like finding an element in an array, etc.
 
 ### parser
 This package implements all functions to parse received messages into structs, and compose messages from structs.
 
 ### security
-This package implements all the checks we do on messages, from timestamp age to validity of hashes and signatures.
+This package implements all the checks we do on messages to ensure they are valid, from timestamp age to validity of hashes and signatures.
```

Comment:
explain CLEARLY the nasty stuff with hash and signature, when its encoded and when its not

Reply:
good idea, go ahead :)